tsl/src/bgw_policy/compression_api.c

```diff
@@ -340,3 +373,76 @@ policy_compression_remove(PG_FUNCTION_ARGS)
 
 	PG_RETURN_BOOL(true);
 }
+
+/* compare cagg job config  with compression job config. If there is an overlap, then
+ * throw an error. We do this since we cannot refresh compressed
+ * regions. We do not want cont. aggregate jobs to fail
+ */
+
+/* If this is a cagg, then mark it as a cagg */
+static Hypertable *
+validate_compress_chunks_hypertable(Cache *hcache, Oid user_htoid, bool *is_cagg)
+{
+	ContinuousAggHypertableStatus status;
+	Hypertable *ht = ts_hypertable_cache_get_entry(hcache, user_htoid, true /* missing_ok */);
+	*is_cagg = false;
+
+	if (ht != NULL)
+	{
+		if (!TS_HYPERTABLE_HAS_COMPRESSION_ENABLED(ht))
+		{
+			ts_cache_release(hcache);
```

Comment:
Why do we need all those `ts_cache_release()` statements? Don't the exceptions clean up all the process state?

Reply:
I also got a bit confused because we register the XactCallback here: https://github.com/timescale/timescaledb/blob/master/src/cache.c#L362

Reply:
IMHO the better we can do here is just remove those `ts_release_cache` lines

Reply:
We have this usage all over the code base. Don't see a reason to add a special comment here